module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1

```diff
@@ -77,13 +77,98 @@ function Unregister-EditorCommand {
     }
 }
 
+<#
+.SYNOPSIS
+    Creates new files and opens them in your editor window
+.DESCRIPTION
+    Creates new files and opens them in your editor window
+.EXAMPLE
+    PS > New-EditorFile './foo.ps1'
+    Creates and opens a new foo.ps1 in your editor
+.EXAMPLE
+    PS > Get-Process | New-EditorFile proc.txt
+    Creates and opens a new foo.ps1 in your editor with the contents of the call to Get-Process
+.EXAMPLE
+    PS > Get-Process | New-EditorFile proc.txt -Force
+    Creates and opens a new foo.ps1 in your editor with the contents of the call to Get-Process. Overwrites the file if it already exists
+.INPUTS
+    Path
+    an array of files you want to open in your editor
+    Value
+    The content you want in the new files
+    Force
+    Overwrites a file if it exists
+#>
+function New-EditorFile {
+    [CmdletBinding()]
+    param(
+        [Parameter()]
+        [String[]]
+        [ValidateNotNullOrEmpty()]
+        $Path,
+
+        [Parameter(ValueFromPipeline=$true)]
+        $Value,
+
+        [Parameter()]
+        [switch]
+        $Force
+    )
+
+    begin {
+        $valueList = @()
+    }
+
+    process {
+        $valueList += $Value
+    }
+
+    end {
+        if ($Path) {
+            foreach ($fileName in $Path)
+            {
+                if (-not (Test-Path $fileName) -or $Force) {
+                    New-Item -Path $fileName -ItemType File | Out-Null
+                    $psEditor.Workspace.OpenFile($fileName)
+                    $psEditor.GetEditorContext().CurrentFile.InsertText(($valueList | Out-String))
+                } else {
+                    $PSCmdlet.WriteError( (
+                        New-Object -TypeName System.Management.Automation.ErrorRecord -ArgumentList @(
+                            [System.IO.IOException]"The file '$fileName' already exists.",
+                            'NewEditorFileIOError',
+                            [System.Management.Automation.ErrorCategory]::WriteError,
+                            $fileName) ) )
+                }
+            }
+        } else {
+            $psEditor.Workspace.NewFile()
+            $psEditor.GetEditorContext().CurrentFile.InsertText(($valueList | Out-String))
+        }
+    }
+}
+
 function Open-EditorFile {
-    param([Parameter(Mandatory=$true)]$FilePaths)
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        [ValidateNotNullOrEmpty()]
+        $Path
```

Comment:
As long as this parameter supports this scenario `Get-ChildItem *.ps1 -r -file | Open-EditorFile`, I'm good.

Reply:
So @rkeithhill ... as it turns out, this will proceed to open the file temporarily (with the italicized file name)... so what ends up happening is, it opens the file temporarily and then the next file opens and replaces the file temporarily.

I'm not sure how to durably open files - if that's possible. I'll have to dig around.

@daviwil any thoughts?

Reply:
Ah, yes.  That's why I double-click the editor window tab.  That converts it from a temp (resuable) window to a permanent one.  Not sure how you do that programmatically though.

Reply:
Exactly. Maybe if @SeeminglyScience or @daviwil know? Otherwise, can I convince you to call this out of scope 😄

Reply:
You can use the `vscode.window.showTextDocument()` that takes a TextDocumentShowOptions object. One of the fields of that object is `preview?`.  That should allow you to create each window in a "non-preview" window.  

https://code.visualstudio.com/docs/extensionAPI/vscode-api#_a-nametextdocumentshowoptionsaspan-classcodeitem-id239textdocumentshowoptionsspan

Reply:
Eh... kind of raises more questions though.

1. Should there be a `isPreview` overload to `OpenFile`?

1. Does that translate well to other editors?

1. Should it just be the default?

1. Should we instead create a request for the `workbench.action.keepEditor` VSCode command and have this function send that request only if the host is VSCode?

Reply:
I had this thought too. I have a working version that sends `preview` along with the `OpenFile` message but I'm not sure that's the best approach as new editors pop up with different functionalities.

Reply:
I would expect that if I used this command to open three files (however I specify those three files) that there would be three editor windows open when the command completes (not one editor window with the last file in it).  For now, I think we just specify isPreview as false.  If asked, we could add preview as a parameter later.  I kind of doubt it will be needed.

Reply:
@rkeithhill so you're saying that the change then is in vscode-powershell to always set the isPreview to false?

That's a quick fix and doesn't require a whole lot of extra plumbing. I'm ok with that, if I'm understanding you correctly.

Reply:
That would be my preference.  Later we could update the openFile() method to take an optional "preview" Boolean parameter.

Reply:
> Later we could update the openFile() method to take an optional "preview" Boolean parameter.

What if I already have this working? Is it worth adding it now or saving it for when someone actually asks for that?

Reply:
I'll just push and you can judge whether we should keep it in or not

Reply:
I'd like to have that option, so yeah, if it's working - go for it.

Reply:
@rkeithhill pushed that. Here's the vscode-powershell PR as well:
https://github.com/PowerShell/vscode-powershell/pull/1197